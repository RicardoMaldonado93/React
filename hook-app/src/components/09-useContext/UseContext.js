import { createContext } from 'react';

/*
 ?  Es un High Order Component
 */
export const UserContext = createContext( null );
