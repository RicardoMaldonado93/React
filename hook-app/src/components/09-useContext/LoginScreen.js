import React, { useContext, useEffect } from 'react'
import { UserContext } from './UseContext'

export const LoginScreen = () => {

    const { setUser } = useContext( UserContext );

    const handleAddUser = (e)=>{
        e.preventDefault();

        const newUser = {
            id: 123456,
            name: "Ricardo",
            email: "ricardomaldonado@gmail.com"
        }

        setUser( newUser );
    }

    
    return (
        <>
            <h1>LoginScreen</h1>   
            <hr />

            <button 
                className="btn btn-primary"
                onClick={ ()=>setUser( {
                    id: 123456,
                    name: "Ricardo",
                    email: "ricardomaldonado@gmail.com"
                }) }
            >
                Login
            </button>
        </>
    )
}
