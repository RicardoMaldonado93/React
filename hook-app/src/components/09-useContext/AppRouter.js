import React from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
  } from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />

                <div className="container">
                    <Switch>
                        /**
                            Si ponemos la ruta mas generica '\', Router toma por defecto la que se este visualizando 
                            ya que por expresion regular coincide con cualquiera de las rutas, por recomendacion la ruta 
                            generica siempre debe ir al final, o agregar la props 'exact' para que este busque exactamente
                            la ruta seteada en el browser.
                        */

                        <Route exact path="/" component={ HomeScreen } />
                        <Route exact path="/about" component={ AboutScreen } />
                        <Route exact path="/login" component={ LoginScreen } />
                        
                        /** Default Router */
                        <Redirect to="/" />

                    </Switch>
                </div>
                
            </div>
        </Router>
    )
}
