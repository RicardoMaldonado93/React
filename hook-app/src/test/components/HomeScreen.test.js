import { mount, shallow } from "enzyme";
import { HomeScreen } from "../../components/09-useContext/HomeScreen";
import React from 'react';
import { UserContext } from "../../components/09-useContext/UseContext";

describe('Unitary test with <HomeScreen />', () => {
    
    const user ={
        user: "Ricardo",
        email: "demo@demo.com"
    }

    const wrapper = mount(
        <UserContext.Provider value={{ user }}>
            <HomeScreen /> 
        </UserContext.Provider>
    );

    test('should be shown correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();
    });
    
})
