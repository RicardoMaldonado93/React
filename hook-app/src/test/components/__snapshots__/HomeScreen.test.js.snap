// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unitary test with <HomeScreen /> should be shown correctly 1`] = `
Array [
  <h1>
    HomeScreen
  </h1>,
  <hr />,
  <pre>
    {
   "user": "Ricardo",
   "email": "demo@demo.com"
}
  </pre>,
]
`;
